src/main/java/fi/aalto/cs/apluscourses/intellij/model/SettingsImporterImpl.java

```diff
@@ -82,6 +85,16 @@ public void importProjectSettings(@NotNull Project project, @NotNull Course cour
     File settingsZip = FileUtilRt.createTempFile(project.getName() + "-settings", ".zip");
     CoursesClient.fetchZip(settingsUrl, settingsZip);
     ZipFile zipFile = new ZipFile(settingsZip);
+
+    extractZipTo(zipFile, settingsPath);
+
+    Path workspaceXmlPath = settingsPath.resolve("workspace.xml");
+    Document workspaceXml = createCustomWorkspaceXml(workspaceXmlPath);
+    DomUtil.writeDocumentToFile(workspaceXml, workspaceXmlPath.toFile());
+  }
+
+  private static void extractZipTo(@NotNull ZipFile zipFile, @NotNull Path target)
+      throws IOException {
     List<String> fileNames = zipFile
         .getFileHeaders()
```

Comment:
is there a chance `getFileHeaders()` return `null`?

Reply:
I will check :+1:

Reply:
I checked, and it returns an empty list if the ZIP file doesn't exist, so this is safe